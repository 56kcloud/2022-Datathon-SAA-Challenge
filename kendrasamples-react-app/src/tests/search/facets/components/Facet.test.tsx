import React from "react";
import { Facet } from "../../../../search/facets/components/Facet";
import { shallow } from "enzyme";
import { mockDocumentAttributeValueCountPairs } from "../../../mock/mock";
import _ from "lodash";

const mockFacetValues = mockDocumentAttributeValueCountPairs.map(value => ({
  ...value,
  ValueLabel: value.DocumentAttributeValue.StringValue,
}));

const facetConfiguration = {
    facetsToShowWhenUncollapsed: 5,
    showCount: true,
};

describe("A <Facet />", () => {
  it("renders correctly", () => {
    const component = shallow(
      <Facet
        attributeName="mockAttributeName"
        values={mockFacetValues}
        disableAdd={false}
        facetConfiguration={facetConfiguration}
        onSelectionChange={_.noop}
        onClear={_.noop}
      />
    );

    expect(component).toMatchSnapshot();
  });

  it("renders _source_uri correctly", () => {
    const component = shallow(
      <Facet
        attributeName="_source_uri"
        values={mockFacetValues}
        disableAdd={false}
        facetConfiguration={facetConfiguration}
        onSelectionChange={_.noop}
        onClear={_.noop}
      />
    );

    expect(component).toMatchSnapshot();
  });

  it("renders collapsed correctly", () => {
    const component = shallow(
      <Facet
        attributeName="mockAttributeName"
        values={mockFacetValues}
        disableAdd={false}
        facetConfiguration={{
            ...facetConfiguration,
            facetsToShowWhenUncollapsed: 1,
        }}
        onSelectionChange={_.noop}
        onClear={_.noop}
      />
    );

    expect(component).toMatchSnapshot();
  });

  it("renders expanded correctly", () => {
    const component = shallow<Facet>(
      <Facet
        attributeName="mockAttributeName"
        values={mockFacetValues}
        disableAdd={false}
        facetConfiguration={{
            ...facetConfiguration,
            facetsToShowWhenUncollapsed: 1,
        }}
        onSelectionChange={_.noop}
        onClear={_.noop}
      />
    );

    component.instance().showMore();

    expect(component).toMatchSnapshot();
  });

  it("renders count hidden correctly", () => {
    const component = shallow(
      <Facet
        attributeName="mockAttributeName"
        values={mockFacetValues}
        disableAdd={false}
        facetConfiguration={{
            ...facetConfiguration,
            showCount: false,
        }}
        onSelectionChange={_.noop}
        onClear={_.noop}
      />
    );

    expect(component).toMatchSnapshot();
  });

  it("renders correctly if disableAdd", () => {
    const component = shallow(
      <Facet
        attributeName="mockAttributeName"
        values={mockFacetValues}
        disableAdd={true}
        facetConfiguration={facetConfiguration}
        onSelectionChange={_.noop}
        onClear={_.noop}
      />
    );

    expect(component).toMatchSnapshot();
  });
});
