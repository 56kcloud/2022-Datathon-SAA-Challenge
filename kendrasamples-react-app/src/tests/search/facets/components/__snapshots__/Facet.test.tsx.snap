// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A <Facet /> renders _source_uri correctly 1`] = `
<div
  className="facet"
>
  <div
    className="facet-title"
  >
    Source URI
     
  </div>
  <div
    className="facet-value-container"
  >
    <FacetCheckbox
      attributeName="_source_uri"
      count={10}
      disableAdd={false}
      facetConfiguration={
        Object {
          "facetsToShowWhenUncollapsed": 5,
          "showCount": true,
        }
      }
      key="0"
      onSelectionChange={[Function]}
      selected={false}
      value={
        Object {
          "StringValue": "mockStringValue1",
        }
      }
      valueLabel="mockStringValue1"
    />
    <FacetCheckbox
      attributeName="_source_uri"
      count={20}
      disableAdd={false}
      facetConfiguration={
        Object {
          "facetsToShowWhenUncollapsed": 5,
          "showCount": true,
        }
      }
      key="1"
      onSelectionChange={[Function]}
      selected={false}
      value={
        Object {
          "StringValue": "mockStringValue2",
        }
      }
      valueLabel="mockStringValue2"
    />
  </div>
  <div
    className="facet-footer"
  />
</div>
`;

exports[`A <Facet /> renders collapsed correctly 1`] = `
<div
  className="facet"
>
  <div
    className="facet-title"
  >
    mockAttributeName
     
  </div>
  <div
    className="facet-value-container"
  >
    <FacetCheckbox
      attributeName="mockAttributeName"
      count={10}
      disableAdd={false}
      facetConfiguration={
        Object {
          "facetsToShowWhenUncollapsed": 1,
          "showCount": true,
        }
      }
      key="0"
      onSelectionChange={[Function]}
      selected={false}
      value={
        Object {
          "StringValue": "mockStringValue1",
        }
      }
      valueLabel="mockStringValue1"
    />
  </div>
  <div
    className="facet-footer"
  >
    <span
      className="action-link"
      onClick={[Function]}
    >
      Show more...
    </span>
  </div>
</div>
`;

exports[`A <Facet /> renders correctly 1`] = `
<div
  className="facet"
>
  <div
    className="facet-title"
  >
    mockAttributeName
     
  </div>
  <div
    className="facet-value-container"
  >
    <FacetCheckbox
      attributeName="mockAttributeName"
      count={10}
      disableAdd={false}
      facetConfiguration={
        Object {
          "facetsToShowWhenUncollapsed": 5,
          "showCount": true,
        }
      }
      key="0"
      onSelectionChange={[Function]}
      selected={false}
      value={
        Object {
          "StringValue": "mockStringValue1",
        }
      }
      valueLabel="mockStringValue1"
    />
    <FacetCheckbox
      attributeName="mockAttributeName"
      count={20}
      disableAdd={false}
      facetConfiguration={
        Object {
          "facetsToShowWhenUncollapsed": 5,
          "showCount": true,
        }
      }
      key="1"
      onSelectionChange={[Function]}
      selected={false}
      value={
        Object {
          "StringValue": "mockStringValue2",
        }
      }
      valueLabel="mockStringValue2"
    />
  </div>
  <div
    className="facet-footer"
  />
</div>
`;

exports[`A <Facet /> renders correctly if disableAdd 1`] = `
<div
  className="facet"
>
  <div
    className="facet-title"
  >
    mockAttributeName
     
  </div>
  <div
    className="facet-value-container"
  >
    <FacetCheckbox
      attributeName="mockAttributeName"
      count={10}
      disableAdd={true}
      facetConfiguration={
        Object {
          "facetsToShowWhenUncollapsed": 5,
          "showCount": true,
        }
      }
      key="0"
      onSelectionChange={[Function]}
      selected={false}
      value={
        Object {
          "StringValue": "mockStringValue1",
        }
      }
      valueLabel="mockStringValue1"
    />
    <FacetCheckbox
      attributeName="mockAttributeName"
      count={20}
      disableAdd={true}
      facetConfiguration={
        Object {
          "facetsToShowWhenUncollapsed": 5,
          "showCount": true,
        }
      }
      key="1"
      onSelectionChange={[Function]}
      selected={false}
      value={
        Object {
          "StringValue": "mockStringValue2",
        }
      }
      valueLabel="mockStringValue2"
    />
  </div>
  <div
    className="facet-footer"
  />
</div>
`;

exports[`A <Facet /> renders count hidden correctly 1`] = `
<div
  className="facet"
>
  <div
    className="facet-title"
  >
    mockAttributeName
     
  </div>
  <div
    className="facet-value-container"
  >
    <FacetCheckbox
      attributeName="mockAttributeName"
      count={10}
      disableAdd={false}
      facetConfiguration={
        Object {
          "facetsToShowWhenUncollapsed": 5,
          "showCount": false,
        }
      }
      key="0"
      onSelectionChange={[Function]}
      selected={false}
      value={
        Object {
          "StringValue": "mockStringValue1",
        }
      }
      valueLabel="mockStringValue1"
    />
    <FacetCheckbox
      attributeName="mockAttributeName"
      count={20}
      disableAdd={false}
      facetConfiguration={
        Object {
          "facetsToShowWhenUncollapsed": 5,
          "showCount": false,
        }
      }
      key="1"
      onSelectionChange={[Function]}
      selected={false}
      value={
        Object {
          "StringValue": "mockStringValue2",
        }
      }
      valueLabel="mockStringValue2"
    />
  </div>
  <div
    className="facet-footer"
  />
</div>
`;

exports[`A <Facet /> renders expanded correctly 1`] = `
<div
  className="facet"
>
  <div
    className="facet-title"
  >
    mockAttributeName
     
  </div>
  <div
    className="facet-value-container"
  >
    <FacetCheckbox
      attributeName="mockAttributeName"
      count={10}
      disableAdd={false}
      facetConfiguration={
        Object {
          "facetsToShowWhenUncollapsed": 1,
          "showCount": true,
        }
      }
      key="0"
      onSelectionChange={[Function]}
      selected={false}
      value={
        Object {
          "StringValue": "mockStringValue1",
        }
      }
      valueLabel="mockStringValue1"
    />
    <FacetCheckbox
      attributeName="mockAttributeName"
      count={20}
      disableAdd={false}
      facetConfiguration={
        Object {
          "facetsToShowWhenUncollapsed": 1,
          "showCount": true,
        }
      }
      key="1"
      onSelectionChange={[Function]}
      selected={false}
      value={
        Object {
          "StringValue": "mockStringValue2",
        }
      }
      valueLabel="mockStringValue2"
    />
  </div>
  <div
    className="facet-footer"
  >
    <span
      className="action-link"
      onClick={[Function]}
    >
      Show less...
    </span>
  </div>
</div>
`;
